pkg/client/client.go

```diff
@@ -823,6 +823,35 @@ func (c *KpmClient) DownloadFromGit(dep *pkg.Git, localPath string) (string, err
 	return localPath, err
 }
 
+// Parse kcl.mod file and extract dependencies
+func (c *KpmClient) ParseKclModFile(kclPkg *pkg.KclPkg) (map[string][]string, error) {
```

Comment:
Why do we need to analyze dependency graph 2 line by line instead of directly using the corresponding fields.

Reply:
Parsing the `kcl.mod `file line by line allows for flexibility in handling different file formats and variations, it allows us to handle additional metadata .

Reply:
How about this function? https://github.com/kcl-lang/kpm/blob/main/pkg/package/package.go#L38

Reply:
`ParseKclModFile` This function parses the `kcl.mod `file line by line to extract dependencies. It reads the file, splits it into lines, then extracts the dependency name and version from each line, storing them in a map where the dependency name is the key and the versions are stored as a slice of strings.
`dependency_graph.go`
`ConstructDependencyGraphFromModFile` This function utilizes ParseKclModFile to obtain the dependency map and then constructs a dependency graph using ConstructDependencyGraph.

Reply:
The `kcl.mod` file is a `toml` format file, and I didn't see the corresponding test cases. Can you give some examples? In addition, on Windows systems, file wrapping may be caused by `\r\n`

Reply:
Yes , I can modify the function to handle these concerns...

Reply:
Thank you! You can update the code and add more unit test cases in this PR.